src/TraceParserGen.Tests/ParserGenerationTests.cs

```diff
@@ -0,0 +1,288 @@
+            {
+                Output.WriteLine("Skipping test on non-Windows platform. TraceParserGen.exe requires .NET Framework.");
+                return;
+            }
+
+            // Arrange
+            string manifestPath = Path.Combine(TestDataDir, "SimpleTest.manifest.xml");
+            string outputCsPath = Path.Combine(OutputDir, "SimpleTestParser.cs");
+            
+            Output.WriteLine($"Manifest: {manifestPath}");
+            Output.WriteLine($"Output: {outputCsPath}");
+
+            Assert.True(File.Exists(manifestPath), $"Manifest file not found: {manifestPath}");
+
+            // Act - Step 1: Run TraceParserGen.exe
+            string traceParserGenPath = GetTraceParserGenExePath();
+            Output.WriteLine($"TraceParserGen.exe: {traceParserGenPath}");
+
+            var exitCode = RunTraceParserGen(traceParserGenPath, manifestPath, outputCsPath);
+
+            // Assert - Step 1: Verify TraceParserGen succeeded
+            Assert.Equal(0, exitCode);
+            Assert.True(File.Exists(outputCsPath), $"Generated C# file not found: {outputCsPath}");
+
+            // Verify the generated file has expected content
+            string generatedContent = File.ReadAllText(outputCsPath);
+            Assert.Contains("class", generatedContent);
+            Assert.Contains("TraceEventParser", generatedContent);
+
+            Output.WriteLine("Successfully generated parser from manifest");
+
+            // Act - Step 2: Create and build a test console application
+            string testProjectDir = Path.Combine(OutputDir, "TestApp");
+            Directory.CreateDirectory(testProjectDir);
+
+            CreateTestConsoleApp(testProjectDir, outputCsPath);
+
+            // Act - Step 3: Build the test application
+            var buildExitCode = BuildTestApp(testProjectDir);
+            Assert.Equal(0, buildExitCode);
+
+            // Act - Step 4: Run the test application
+            var runExitCode = RunTestApp(testProjectDir);
+
+            // Assert - Step 4: Verify test app ran successfully (no crashes, no asserts)
+            Assert.Equal(0, runExitCode);
+
+            Output.WriteLine("Test completed successfully");
+        }
+
+        private int RunTraceParserGen(string exePath, string manifestPath, string outputPath)
+        {
+            ProcessStartInfo startInfo;
+            
+            // On Linux/Mac, we need to use mono to run .NET Framework executables
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                startInfo = new ProcessStartInfo
+                {
+                    FileName = exePath,
+                    Arguments = $"\"{manifestPath}\" \"{outputPath}\"",
+                    UseShellExecute = false,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    CreateNoWindow = true
+                };
+            }
+            else
```

Comment:
The tests only run on Windows, so you can remove the else here and just unconditionally run the Windows path.